src/main/java/org/elasticsearch/index/query/QueryParseContext.java

```diff
@@ -285,9 +285,8 @@ public void addInnerHits(String name, InnerHitsContext.BaseInnerHits context) {
         innerHitsContext.addInnerHitDefinition(name, context);
     }
 
-    @Nullable
-    public Query parseInnerQuery() throws IOException, QueryParsingException {
-        // move to START object
+    public QueryBuilder toQueryBuilder() throws IOException {
```

Comment:
not sure I am happy with the name of this method. I don't really like the Builder suffix, although it does return (temporarily?) a builder. Also, it does parse something, so using parse seems a good idea here? javadocs would be good too.